[Bug] Hitting a C++ error on MacOS

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

doris/be/src/io/cache/block/block_file_cache_settings.h:29:33: error: constexpr variable cannot have non-literal type 'const std::array<size_t, 4>' (aka 'const array<unsigned long, 4>')
constexpr std::array<size_t, 4> percentage {17, 2, 1, 20};

MacOSX12.3.sdk/usr/include/c++/v1/__tuple:219:64: note: template is declared here
template <class _Tp, size_t _Size> struct _LIBCPP_TEMPLATE_VIS array;

### What You Expected?

编译通过

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/src/io/cache/block/block_file_cache_settings.h
+++ b/be/src/io/cache/block/block_file_cache_settings.h
@@ -20,6 +20,8 @@
 
 #pragma once
 
+#include <array>
+
 #include "io/cache/block/block_file_cache_fwd.h"
 
 namespace doris {

